pkgs/development/tools/volta/default.nix

```
@@ -0,0 +1,48 @@
+{
+  lib,
+  rustPlatform,
+  libiconv,
+  stdenv,
+  installShellFiles,
+  darwin,
+  fetchFromGitHub,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "volta";
+  version = "1.1.1";
+
+  src = fetchFromGitHub {
+    owner = "volta-cli";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-+j3WRpunV+3YfZnyuKA/CsiKr+gOaP2NbmnyoGMN+Mg=";
+  };
+
+  cargoHash = "sha256-g6Ai0QN9cDiNX/YHdMM7/llJ/pCWZtv66wTaKvuN+bE=";
+
+  buildInputs = [installShellFiles]
+  ++ lib.optionals stdenv.isDarwin [darwin.apple_sdk.frameworks.Security libiconv];
+
+  HOME = "$TMPDIR";
+
+  postInstall = ''
+    installShellCompletion --cmd volta \
+      --bash <($out/bin/volta completions bash) \
+      --fish <($out/bin/volta completions fish) \
+      --zsh <($out/bin/volta completions zsh)
+  '';
+  meta = with lib; {
+    description = "The Hassle-Free JavaScript Tool Manager";
+    longDescription = ''
+      With Volta, you can select a Node engine once and then stop worrying
+      about it. You can switch between projects and stop having to manually
+      switch between Nodes. You can install npm package binaries in your
+      toolchain without having to periodically reinstall them or figure out why
+      they’ve stopped working.
+    '';
+    homepage = "https://volta.sh/";
+    changelog = "https://github.com/volta-cli/volta/blob/main/RELEASES.md";
+    license = with licenses; [bsd2];
+    maintainers = with maintainers; [fbrs];
```

```suggestion
    license = with licenses; [ bsd2 ];
    maintainers = with maintainers; [ fbrs ];
```
don't use alejandra. It is very inconsistent with itself and adds a lot of extra space in the beginning of the file while clumsing things here together.

---

I like that it's so opinionated but I'll make sure to not use it in `nixpkgs` then